infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -445,3 +494,95 @@ private fun verifySwitchSuggestions(
+        .mapNotNull { (unusedDuplicate, alignment) ->
+            getAlignmentStartAndEndM(geocodingContext, alignment)?.let { startAndEnd ->
+                unusedDuplicate to startAndEnd
+            }
+        }
+
+    val geocodedSplitTargets = splitTargetLocationTracks
+        .mapNotNull { (locationTrack, alignment) ->
+            getAlignmentStartAndEndM(geocodingContext, alignment)?.let { startEnd ->
+                locationTrack to startEnd
+            }
+        }
+
+    return geocodedUnusedDuplicates.map { (duplicate, duplicateStartAndEnd) ->
+        geocodedSplitTargets
+            .fold(
+                MostOverlappingLocationTrackReference()
+            ) { currentBest, (splitTarget, splitTargetStartEnd) ->
+
+                if (currentBest.overlapPercentage > 99.9) {
+                    currentBest
+                } else {
+                    val overlapStart = maxOf(duplicateStartAndEnd.start, splitTargetStartEnd.start)
+                    val overlapEnd = minOf(duplicateStartAndEnd.end, splitTargetStartEnd.end)
+
+                    val overlap = maxOf(0.0, overlapEnd - overlapStart)
+                    val intervalLength = duplicateStartAndEnd.end - duplicateStartAndEnd.start
+
+                    val overlapPercentage = overlap / intervalLength * 100
+                    if (overlapPercentage > currentBest.overlapPercentage) {
+                        MostOverlappingLocationTrackReference(
+                            locationTrack = splitTarget,
+                            overlapPercentage = overlapPercentage,
+                        )
+                    } else {
+                        currentBest
+                    }
+                }
+
+            }.let { bestNewLocationTrackReference ->
+                val newReferenceTrackId = bestNewLocationTrackReference.locationTrack?.id as IntId?
+                checkNotNull(newReferenceTrackId) {
+                    "Could not find a new location track reference for an unused duplicate!"
+                }
+
+                duplicate.copy(duplicateOf = newReferenceTrackId)
+            }
+    }
+}
+
+private data class AlignmentStartAndEndMeters(
+    val start: Double,
+    val end: Double,
+)
+
+private fun getAlignmentStartAndEndM(
+    geocodingContext: GeocodingContext,
+    alignment: LayoutAlignment
+): AlignmentStartAndEndMeters? {
+    val alignmentStart = alignment.start
+    val alignmentEnd = alignment.end
+
+    return if (alignmentStart != null && alignmentEnd != null) {
+        val startMeters = geocodingContext.getM(alignmentStart)?.first
+        val endMeters = geocodingContext.getM(alignmentEnd)?.first
+
+        when {
+            startMeters == null -> null
+            endMeters == null -> null
+
+            else -> AlignmentStartAndEndMeters(
+                start = startMeters,
+                end = endMeters,
+            )
+        }
+    } else {
+        null
+    }
+}
```

Comment:
Olikohan jotain vastaavaa jo jossakin muualla olemassa/käytetty?

Reply:
Ei varmaankaan. Ei taida loppujen lopuksi olla kovin yleistä olla kiinnostunut pituusmittauslinjan m-arvosta. Tuo sopisi kyllä ihan hyvin tuonne geokoodauskontekstiinkin jossain muodossa, vaikkapa `getMRange(startPoint, endPoint): ClosedRange<Double>`. Toki ehkä vähän hähmäinen koska 2 pistettä voi olla mitenpäin vain - pitäisikö tuon heittää virhe jos ne on pituusmittauslinjan menosuuntaan väärin päin vai pitäisikö ehkä vain palauttaa väli jos se on jotenkinpäin tulkittavissa? Eli sinällään varmaan ok täällä spesifinä juttuna jossa kontekstista tiedetään että pitäisi olla saman linjan raide -> samoin päin. 

Itse koodiin: tämä olisi minusta luettavampi (joskaan ei oikosulje ihan yhtä tehokkaasti) jotensakin tyyliin:
```
val startMeters = alignment.start?.let(geocodingContext::getM)?.first
val endMeters = alignment.end?.let(geocodingContext::getM)?.first
return if (startMeters != null && endMeters != null) {
  AlignmentStartAndEndMeters(startMeters, endMeters)
} else {
  null
}
```